### ISSUE TITLE
Consider caching the XmlSerializers in the EdiHelper class

### ISSUE THREAD

**Issue by clafollett** [DonZoeggerle/ediFabric#24]
The ediFabric library has been great for working with EDIFACT files. One thing we have noticed is their appears to be a bit of overhead in Serializing and Deserializing messages. Upon review of the source, it appears the EdiHelper class creates a new XmlSerializer for each call to EdiHelper.Serialize<T> and EdiHelper.Deserialize<T>. There is a bit of overhead in creating a new XmlSerializer especially considering the size of the EDIFACT schemas.

I recommend creating a serialization helper which can keep a static list of serializers and only creates one for a given message type upon first request and returns an existing one if it exists.

Here is a potential helper class you can use:

```
public static class SerializationHelper
{
	private readonly static object SearializerLock = new object();
	private readonly static Dictionary<string, XmlSerializer> SerializerCache = Dictionary<string, XmlSerializer>();

	public static XmlSerializer GetSerializerInstance<T>() where T : class
	{
		var key = (typeof(T)).FullName;

		if (SerializerCache.ContainsKey(key))
			return SerializerCache[key];

		lock (SearializerLock)
		{
			// Double check lock to make sure another thread did not create one already
			if (SerializerCache.ContainsKey(key))
				return SerializerCache[key];

			SerializerCache[key] = new XmlSerializer(typeof(T));
		}

		return SerializerCache[key];
	}
}
```

**Comment by DonZoeggerle** [comment]
Thank you for the suggestion, it does make sense to cache the serializers and I will add this in the next release.

**Comment by DonZoeggerle** [comment]
Closing the issue - resolved in version 6.7.5
